packages/components/src/Popover/Popover.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useRef } from "react";
+import {
+  RenderResult,
+  act,
+  cleanup,
+  fireEvent,
+  render,
+} from "@testing-library/react";
+import { Popover } from ".";
+import { PopoverProps } from "./Popover";
+
+afterEach(cleanup);
+
+let rendered: RenderResult;
+const content = "Test Content";
+
+const PopoverTestComponent = (props: PopoverProps) => {
+  const checkboxRef = useRef();
+  return (
+    <>
+      <div ref={checkboxRef}></div>
+      <Popover {...props} attachTo={checkboxRef}>
+        {props.children}
+      </Popover>
+    </>
+  );
+};
+
+it("should render a Popover with the content and dismiss button", async () => {
+  const handleClose = jest.fn();
+
+  await act(async () => {
+    rendered = render(
+      <PopoverTestComponent
+        attachTo={undefined}
+        open={true}
+        onRequestClose={handleClose}
+      >
+        {content}
+      </PopoverTestComponent>,
+    );
+  });
+
+  const { queryByText, queryByTestId, container } = rendered;
+
+  expect(queryByText(content)).toBeTruthy();
```

Comment:
Might be nicer to test that we're actually getting an HTML element here.

```suggestion
  expect(queryByText(content)).toBeInstanceOf(HTMLElement);
```